src/data/deleteChannel.js

```diff
@@ -0,0 +1,43 @@
+const debug = require('debug')('appstore:server:data:deleteChannel')
+/**
+ * Deletes a channel if no apps is published to it, otherwise it will throw an error.
+ *
+ * @param {string} uuid UUID for the channel to delete
+ * @param {object} knex database connection
+ * @param {object} transaction database transaction
+ * @returns {Promise} object
+ */
+module.exports = async (uuid, knex, transaction) => {
+    if (!uuid || !knex) {
+        throw new Error('Invalid parameters, either uuid or knex is missing')
+    }
+
+    try {
+        const [{ count }] = await knex('apps_view')
+            .where('channel_id', uuid)
+            .count()
+        if (+count > 0) {
+            return {
+                deleted: 0,
+                success: false,
+                message:
+                    'Can not delete a channel which has apps published to it.',
+            }
+        }
```

Comment:
Instead of doing this you can throw any of the errors from db-errors (see [here ](https://github.com/dhis2/app-hub/blob/master/src/plugins/errorMapper.js#L15), ie. an CheckViolationError, or you can create a custom one). This would remove the need for this object-api, and you could just return the count or something. In the try-catch you would need to rollback and then just rethrow the error as the errormapper will handle the rest.